### PR TITLE
Replace org.junit.Assert with assertj

### PR DESCRIPTION
**Pull request by schlosna**
**Goals (and why)**:
Replace org.junit.Assert with assertj to consistently prefer AssertJ. This moves atlasdb one step closer to junit5 and parallelized tests.

**Implementation Description (bullets)**:
Manually refactoring as [baseline assertj automation](https://github.com/palantir/assertj-automation) did not migrate these remaining uses.

**Testing (What was existing testing like?  What have you done to improve it?)**:
CI tests

**Concerns (what feedback would you like?)**:
Lots of manual changes to test code, potential for errors (fortunately tests catch most of this).

**Where should we start reviewing?**:
I tried to split up commits by type of cleanup

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

File: atlasdb-dbkvs-tests/src/test/java/com/palantir/nexus/db/pool/HikariCpConnectionManagerTest.java [hunk 1]
```diff
@@ -67,35 +63,42 @@ public void testCanGetConnection() throws SQLException {
...
     public void testCantGetConnectionIfClosed() throws SQLException {
         manager.close();
-        thrown.expect(SQLException.class);
-        thrown.expectMessage("Hikari connection pool already closed!");
-        try (Connection conn = manager.getConnection()) {
-            fail("fail");
-        }
+        assertThatThrownBy(() -> {
+                    try (Connection connection = manager.getConnection()) {
+                        assertThat(connection).isNotNull();
```
Comment (Jolyon-S) [inline atlasdb-dbkvs-tests/src/test/java/com/palantir/nexus/db/pool/HikariCpConnectionManagerTest.java:68]: This is a bit strange - are we asserting that the `manager.getConnection` method fails, but at the same time want to clean up the resources? Can we rewrite this some other way?

File: atlasdb-dbkvs-tests/src/test/java/com/palantir/nexus/db/pool/HikariCpConnectionManagerTest.java [hunk 2]
```diff
@@ -67,35 +63,42 @@ public void testCanGetConnection() throws SQLException {
...
+                    }
+                })
+                .isInstanceOf(SQLException.class)
+                .hasMessageContaining("Hikari connection pool already closed!");
     }
 
     @Test
     public void testCantGetConnectionIfPoolExhausted() throws SQLException {
+        //noinspection unused - try-with-resources gets & closes connections
```
Comment (Jolyon-S) [inline atlasdb-dbkvs-tests/src/test/java/com/palantir/nexus/db/pool/HikariCpConnectionManagerTest.java:77]: Same here

File: atlasdb-commons/src/test/java/com/palantir/common/concurrent/InterruptibleFutureTest.java [hunk 3]
```diff
@@ -89,10 +79,10 @@ public void testCancelTrueWithSleep() throws Exception {
...
         interruptible.started.await();
         interruptible.cancel(true);
-        expectedException.expect(Matchers.<Throwable>either(instanceOf(CancellationException.class))
-                .or(Matchers.<Throwable>both(instanceOf(ExecutionException.class))
-                        .and(ThrowableCauseMatcher.hasCause(instanceOf(InterruptedException.class)))));
-        interruptible.get();
+        assertThatThrownBy(interruptible::get)
+                .isInstanceOfAny(CancellationException.class, ExecutionException.class)
+                .getRootCause()
+                .isInstanceOf(InterruptedException.class);
```
Comment (Jolyon-S) [inline atlasdb-commons/src/test/java/com/palantir/common/concurrent/InterruptibleFutureTest.java:85]: isn't there `hasRootCauseInstanceOf`?

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/SweepBatchConfigTest.java [hunk 4]
```diff
@@ -15,57 +15,54 @@
...
-    public ExpectedException exception = ExpectedException.none();
 
     @Test
     public void canCreateConfig() {
-        ImmutableSweepBatchConfig.builder()
-                .maxCellTsPairsToExamine(1)
-                .candidateBatchSize(1)
-                .deleteBatchSize(1)
-                .build();
+        assertThat(ImmutableSweepBatchConfig.builder()
```
Comment (Jolyon-S) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/SweepBatchConfigTest.java:28]: there is also
```java
assertThatCode(() -> ...)
    .doesNotThrowAnyException();
```

which may be more fitting for places like this (although checking it isn't null isn't wrong, exactly)

File: timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceTests.java [hunk 5]
```diff
@@ -56,22 +52,21 @@ public void shouldLimitRequestsForMoreThanTenThousandTimestamps() {
...
-        exception.expect(RuntimeException.class);
-        exception.expectCause(is(failure));
-
         timestampBoundStore.failWith(failure);
-
-        getTimestampService().getFreshTimestamp();
+        assertThatThrownBy(() -> getTimestampService().getFreshTimestamp())
+                .isInstanceOf(RuntimeException.class)
+                .getCause()
+                .isEqualTo(failure);
```
Comment (Jolyon-S) [inline timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceTests.java:69]: again, can't you do `hasCause(failure)` (or something like that)?

File: atlasdb-client/src/test/java/com/palantir/atlasdb/schema/stream/StreamStoreDefinitionBuilderTest.java [hunk 6]
```diff
@@ -15,22 +15,27 @@
...
 
     @Test
     public void testMaxInMemoryThreshold() {
-        new StreamStoreDefinitionBuilder("test", "test", ValueType.VAR_LONG)
-                .inMemoryThreshold(StreamStoreDefinition.MAX_IN_MEMORY_THRESHOLD)
-                .build();
+        assertThat(new StreamStoreDefinitionBuilder("test", "test", ValueType.VAR_LONG)
+                        .inMemoryThreshold(StreamStoreDefinition.MAX_IN_MEMORY_THRESHOLD)
+                        .build())
+                .isNotNull();
```
Comment (Jolyon-S) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/schema/stream/StreamStoreDefinitionBuilderTest.java:39]: again with `assertThatCode`

**Review by Jolyon-S — COMMENTED** [review]
Largely looks good - thank you very much for tackling this! A few comments on possibly nicer ways to express things (although I don't think it particularly affects test behaviour).

**Comment by schlosna** [comment]
Thanks for the review, comments addressed

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java [hunk 7]
```diff
@@ -329,18 +332,21 @@ public void testConcurrentWriteSkewCell() throws InterruptedException, BrokenBar
...
-        }
+        assertThat(catchThrowable(() -> {
+                    t2.commit();
+                    future.get();
+                }))
+                .satisfiesAnyOf(
+                        t -> assertThat(t).isInstanceOf(TransactionFailedRetriableException.class), t -> assertThat(t)
+                                .isInstanceOf(ExecutionException.class)
+                                .getCause()
+                                .isInstanceOf(TransactionFailedRetriableException.class));
```
Comment (Jolyon-S) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java:349]: and here

File: timestamp-impl/src/test/java/com/palantir/timestamp/TimestampAllocationFailuresTest.java [hunk 8]
```diff
@@ -83,10 +78,10 @@ public void shouldAllowTryingToIssueMoreTimestampsAfterAServiceNotAvailableExcep
...
         ignoringExceptions(() -> allocationFailures.responseTo(MULTIPLE_RUNNING_SERVICES_FAILURE));
 
-        exception.expectCause(is(MULTIPLE_RUNNING_SERVICES_FAILURE));
-        exception.expect(ServiceNotAvailableException.class);
-
-        allocationFailures.verifyWeShouldIssueMoreTimestamps();
+        assertThatThrownBy(allocationFailures::verifyWeShouldIssueMoreTimestamps)
+                .isInstanceOf(ServiceNotAvailableException.class)
+                .getCause()
+                .isEqualTo(MULTIPLE_RUNNING_SERVICES_FAILURE);
```
Comment (Jolyon-S) [inline timestamp-impl/src/test/java/com/palantir/timestamp/TimestampAllocationFailuresTest.java:84]: and here

File: atlasdb-config/src/test/java/com/palantir/atlasdb/config/AtlasDbConfigTest.java [hunk 9]
```diff
@@ -205,24 +212,26 @@ public void absentNamespaceRequiresTimelockClient() {
...
     }
 
     @Test
     public void namespaceAcceptsEmptyKvsNamespaceAndTimelockClient() {
-        AtlasDbConfig config = ImmutableAtlasDbConfig.builder()
-                .namespace("a client")
-                .keyValueService(KVS_CONFIG_WITHOUT_NAMESPACE)
-                .timelock(TIMELOCK_CONFIG_WITH_OPTIONAL_EMPTY_CLIENT)
-                .build();
+        assertThat(ImmutableAtlasDbConfig.builder()
```
Comment (Jolyon-S) [inline atlasdb-config/src/test/java/com/palantir/atlasdb/config/AtlasDbConfigTest.java:229]: this should be `assertThatCode`

**Review by Jolyon-S — APPROVED** [review]
Sorry, I missed a couple of possible improvements in my pass. Can you do a quick check (cmd+f) that there are no others that I missed when you action these? I'll approve,  but please action these before merging.

**Comment by schlosna** [comment]
Follow up comments on https://github.com/palantir/atlasdb/pull/5540

**Pull request merged**